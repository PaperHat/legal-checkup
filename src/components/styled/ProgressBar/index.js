import styled from 'styled-components'

const ProgressBar = styled.ul`
  list-style: none;
  display: flex;
  background-color: gainsboro;
  margin: 0 0.2rem;
  padding: 0;
`

ProgressBar.displayName = 'StyledProgressBar'

export default ProgressBar
