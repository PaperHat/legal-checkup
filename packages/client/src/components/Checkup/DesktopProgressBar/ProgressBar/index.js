import styled from 'styled-components'

const ProgressBar = styled.nav`
  background-color: #fff;
  display: flex;
  justify-content: center;
  margin: 20px 0;
`

ProgressBar.displayName = 'StyledDesktopProgressBar'

export default ProgressBar
