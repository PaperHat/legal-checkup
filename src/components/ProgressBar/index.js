import React from 'react'
import { times } from 'ramda'

import { Question } from '..'
import { StyledProgressBar } from '../styled'

function makeQuestions (answered, current, total) {
  return times(idx => {
    const question = ++idx

    return (
      <Question
        key={question}
        answered={answered}
        current={current}
        question={question}
      >
        {question}
      </Question>
    )
  }, total)
}

export default function ProgressBar ({ answered = 2, current = 2, total = 3 }) {
  return (
    <StyledProgressBar>
      {makeQuestions(answered, current, total)}
    </StyledProgressBar>
  )
}
