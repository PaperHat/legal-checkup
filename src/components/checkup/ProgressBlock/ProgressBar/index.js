import styled from 'styled-components'
import styledMap from 'styled-map'

import { DESKTOP, MOBILE, TABLET } from '../../../../constants'

const ProgressBar = styled.nav`
  background-color: #eeedea;
  display: flex;
  width: 100%;
  flex-wrap: nowrap;
  align-items: center;
  justify-content: center;
  letter-spacing: normal;
  text-align: center;
  margin: 20px 0;

  ${styledMap('format', {
    [DESKTOP]: ``,
    [MOBILE]: ``,
    [TABLET]: ``
  })}
`

ProgressBar.displayName = 'ProgressBar'

export default ProgressBar
