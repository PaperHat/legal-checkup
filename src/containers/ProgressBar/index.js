import { connect } from 'react-redux'

import { ProgressBar } from '../../components'
import {
  getActiveQuestion,
  getQuestions,
  getResponses,
  questionSelected,
  previousQuestion,
  nextQuestiion
} from '../../state'

function mapStateToProps (state) {
  return {
    activeQuestion: getActiveQuestion(state),
    questions: getQuestions(state),
    responses: getResponses(state)
  }
}

function mapDispatchToProps (dispatch) {
  return {
    handleClick: questionNumber => dispatch(questionSelected(questionNumber)),
    handlePrevious: () => dispatch(previousQuestion()),
    handleNext: () => dispatch(nextQuestiion())
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(ProgressBar)
