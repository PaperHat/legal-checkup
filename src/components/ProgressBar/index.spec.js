import React from 'react'
import { shallow } from 'enzyme'

import ProgressBar from '.'

describe('components:ProgressBar', () => {
  it('renders the ProgressBar and CSS properties properly', () => {
    expect(
      toJson(shallow(<ProgressBar answered={2} current={2} total={3} />))
    ).toMatchSnapshot()
  })

  it('renders the ProgressBar and correct number of Question components', () => {
    const total = 4

    expect(
      shallow(<ProgressBar answered={2} current={2} total={total} />).find(
        'Question'
      )
    ).toHaveLength(total)
  })
})
