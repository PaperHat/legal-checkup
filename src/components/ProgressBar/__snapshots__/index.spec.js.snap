// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components:ProgressBar renders the ProgressBar and CSS properties properly 1`] = `
<StyledProgressBar>
  <StyledQuestion
    answered={2}
    current={2}
    key="1"
  >
    1
  </StyledQuestion>
  <StyledQuestion
    answered={2}
    current={2}
    key="2"
  >
    2
  </StyledQuestion>
  <StyledQuestion
    answered={2}
    current={2}
    key="3"
  >
    3
  </StyledQuestion>
</StyledProgressBar>
`;
